src/providers/notification.provider.ts

```diff
@@ -21,6 +23,8 @@ export class NotificationProvider implements Provider<INotification> {
       return this.emailProvider.publish(data);
     } else if (data.type === MessageType.Push && this.pushProvider) {
       return this.pushProvider.publish(data);
+    } else if (data.type === MessageType.SocketIO && this.socketProvider) {
```

Comment:
socket io is not a new message type. Mesage type will remain push. socket io is just a new push provider.